File: app/src/main/java/dev/ragnarok/fenrir/api/model/response/PushSettingsResponse.java
package dev.ragnarok.fenrir.api.model.response;

import static dev.ragnarok.fenrir.util.Objects.nonNull;

import com.google.gson.annotations.SerializedName;

import java.util.Collections;
import java.util.List;

import dev.ragnarok.fenrir.util.Utils;


public class PushSettingsResponse {
    @SerializedName("conversations")
    public ConversationsPush conversations;

    public List<ConversationsPush.ConversationPushItem> getPushSettings() {
        if (nonNull(conversations) && !Utils.isEmpty(conversations.items)) {
            return conversations.items;
        }
        return Collections.emptyList();
    }

    public static class ConversationsPush {
        @SerializedName("items")
        List<ConversationPushItem> items;

        public static class ConversationPushItem {
            @SerializedName("disabled_until")
            public int disabled_until;
            @SerializedName("peer_id")
            public int peer_id;
        }
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/api/model/response/PushSettingsResponse.kt
package dev.ragnarok.fenrir.api.model.response

import com.google.gson.annotations.SerializedName
import dev.ragnarok.fenrir.api.model.response.PushSettingsResponse.ConversationsPush.ConversationPushItem

class PushSettingsResponse {
    @SerializedName("conversations")
    var conversations: ConversationsPush? = null
    val pushSettings: List<ConversationPushItem>
        get() = conversations?.items ?: emptyList()

    class ConversationsPush {
        @SerializedName("items")
        var items: List<ConversationPushItem>? = null

        class ConversationPushItem {
            @SerializedName("disabled_until")
            var disabled_until = 0

            @SerializedName("peer_id")
            var peer_id = 0
        }
    }
}